docs/sources/tempo/configuration/_index.md

```diff
@@ -357,19 +393,66 @@ metrics_generator:
             # the metrics if present.
             [dimensions: <list of string>]
 
-            # Custom labeling of dimensions is possible via a list of maps consisting of
-            # "name" <string>, "source_labels" <list of string>, "join" <string>
-            # "name" appears in the metrics, "source_labels" are the actual
-            # attributes that will make up the value of the label and "join" is the
-            # separator if multiple source_labels are provided
-            [dimension_mappings: <list of map>]
+            # Custom labeling mapping
+            dimension_mappings: <list of label mappings>
+                # The new label name
+              - [name: <string>]
+                # The actual attributes that will make the value of the new label
+                [source_labels: <list of strings>]
+                # The separator used to join multiple `source_labels`
+                [join: <string>]
+
             # Enable traces_target_info metrics
-            [enable_target_info: <bool>]
-            # Drop specific labels from traces_target_info metrics
-            [target_info_excluded_dimensions: <list of string>]
+            [enable_target_info: <bool> | default = false]
+
             # Attribute Key to multiply span metrics
             [span_multiplier_key: <string> | default = ""]
 
+            # List of policies that will be applied to spans for inclusion or exclusion.
+            [filter_policies: <list of filter policies config> | default = []]
+
+            # Drop specific labels from `traces_target_info` metrics
+            [target_info_excluded_dimensions: <list of string>]
+
+        local_blocks:
+
+            # Block configuration
+            block: <Block config>
+
+            # Search configuration
+            search: <Search config>
+
+            # How often to run flush checks (??)
+            [flush_check_period: <duration> | default = 10s]
+
+            # After this idle period, a trace is closed. (??)
+            [trace_idle_period: <duration> | default = 10s]
+
+            # Maximum (???) (lifespan for any given block?)
+            [max_block_duration: <duration> | default = 1m]
+
+            # Maximum block size before it is closed (?)
+            [max_block_bytes: <uint64> | default = 500000000]
+
+            # Maximum (???)
+            [complete_block_timeout: <duration> | default = 1h]
+
+            # (Unused?)
+            # Max number of "live" traces
+            [max_live_traces: <uint64>]
+
+            # Whether server spans should be filtered in or not.
+            # true means keep the server kinds (???)
+            # false means drop the server kinds (???)
+            [filter_server_spans: <bool> | default = true]
```

Comment:
Help with these descriptions please

Reply:
Happy to try to help. What do you mean by server kinds?

Reply:
https://github.com/grafana/tempo/blob/56a0b8c383b0efbe7e12477d36f45c680e8bfa44/modules/generator/processor/localblocks/processor.go#L886-L920

It's the span kind(?) client/server kinds I think.

Reply:
My suggestions would be:

```yaml
            # How often to run flush checks (??)
            [flush_check_period: <duration> | default = 10s]
```
> How often to run the flush loop to cut idle traces and blocks

```yaml
            # After this idle period, a trace is closed. (??)
            [trace_idle_period: <duration> | default = 10s]
```
> Duration after which to consider a trace complete if no spans have been received
(this is copied from the flag definition)

```yaml
            # Maximum (???) (lifespan for any given block?)
            [max_block_duration: <duration> | default = 1m]
```
> Maximum duration which the head block can be appended to before cutting it

```yaml
            # Maximum block size before it is closed (?)
            [max_block_bytes: <uint64> | default = 500000000]
```
> Maximum size of the head block before cutting it

```yaml
            # Maximum (???)
            [complete_block_timeout: <duration> | default = 1h]
```
> Duration to keep blocks in the ingester after they have been flushed

```yaml
            # (Unused?)
            # Max number of "live" traces
            [max_live_traces: <uint64>]
```
> Maximum amount of live traces, if this is exceeded traces will be dropped with reason `live_traces_exceeded`.

```yaml
            # Whether server spans should be filtered in or not.
            # true means keep the server kinds (???)
            # false means drop the server kinds (???)
            [filter_server_spans: <bool> | default = true]
```

This code iterates through all spans and only keeps spans that either:
- have span.kind == SPAN_KIND_SERVER
- don't have a parent span

> If enabled, only parent spans or spans with SpanKind server will be retained

Reply:
Out of curiosity, why only retain parents and `Server` spans by default?
What about other spans or `Client` types?